pkg/cachemanager/cachemanager.go

```diff
@@ -157,20 +159,16 @@ func (c *CacheManager) replaceWatchSet(ctx context.Context) error {
 	newWatchSet.Add(c.gvksToSync.GVKs()...)
 	c.gvksToDeleteFromCache.AddSet(c.watchedSet.Difference(newWatchSet))
 
-	if c.danglingWatches || !c.watchedSet.Equals(newWatchSet) {
-		var err error
-		c.watchedSet.Replace(newWatchSet, func() {
-			// *Note the following steps are not transactional with respect to admission control
-
-			// Important: dynamic watches update must happen *after* updating our watchSet.
-			// Otherwise, the sync controller will drop events for the newly watched kinds.
-			err = c.registrar.ReplaceWatch(ctx, newWatchSet.Items())
-		})
+	var err error
+	c.watchedSet.Replace(newWatchSet, func() {
+		// *Note the following steps are not transactional with respect to admission control
 
-		return err
-	}
+		// Important: dynamic watches update must happen *after* updating our watchSet.
+		// Otherwise, the sync controller will drop events for the newly watched kinds.
+		err = c.registrar.ReplaceWatch(ctx, newWatchSet.Items())
+	})
 
-	return nil
+	return err
```

Comment:
To avoid false `if err == nil` results, we should return nil directly if there is no error, not a typed nil.

https://yourbasic.org/golang/gotcha-why-nil-error-not-equal-nil/

Reply:
Good catch! This may have been an oversight initially.